src/main/java/com/ecore/roles/service/MembershipsService.java

```diff
@@ -0,0 +1,14 @@
+package com.ecore.roles.service;
+
+import com.ecore.roles.exception.ResourceNotFoundException;
+import com.ecore.roles.model.Membership;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface MembershipsService {
+
+    Membership assignRoleToMembership(Membership membership) throws ResourceNotFoundException;
+
+    List<Membership> getMemberships(UUID roleId);
```

Comment:
It could be just called "get" or "getAll", because it's inside of the membership service. 

If this class leads the SRP (SOLID) as it should do, all the developers might know that the default domain handled by this class is the membership one. 

It would be even better, if it was called "getByRoleId" or "getAllByRoleId", because this parameter type is a generic one (UUID) and it's not too intuitive by itself. Add "ByRoleId" in the end fits better in these cases.